src/program.ts

```diff
@@ -999,10 +999,10 @@ export class Program extends DiagnosticEmitter {
     return this.resolver.resolveFunction(<FunctionPrototype>prototype, null);
   }
 
-  /** Requires that a non-generic global function is present and returns it. */
-  private requireFunction(name: string): Function {
+  /** Requires that a global function is present and returns it. */
+  private requireFunction(name: string, typeArguments: Type[] | null = null): Function {
```

Comment:
Story here is that I initially experimented with actual `isTrueish<T>` and `isFalseish<T>` internal functions needing this. I'm still not sure if that would be better as it can potentially reduce code size but has the disadvantage that these wouldn't inline well in `-O3s` similar to `isNaN`. Conceptionally, one could argue that we only have `isNaN` because that's an actual stdlib function. Or one could argue that we have it for code size reasons as well, which makes it a bit ambiguous.